src/main/java/com/artipie/maven/util/AutoCloseablePath.java

```
@@ -0,0 +1,120 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.artipie.maven.util;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Adapts {@link Path} to {@link AutoCloseable} interface, quietly deleting on close.
+ * @since 0.1
+ */
+public final class AutoCloseablePath implements AutoCloseable {
+
+    /**
+     * Class logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(AutoCloseablePath.class);
+
+    /**
+     * Actual path.
+     */
+    private final Path path;
+
+    /**
+     * All arg constructor.
+     * @param path A path to instance to wrap over.
+     */
+    public AutoCloseablePath(final Path path) {
+        this.path = path;
+    }
+
+    @Override
+    public void close() throws Exception {
+        try {
+            Files.deleteIfExists(this.path);
+        } catch (final IOException ex) {
+            LOG.warn(String.format("with %s", this.path), ex);
```

@Nimmaster why to hide exception here? Let's throw it

---

To make the application more fault tolerant. It should not fail just because it was unable to delete a file.